src/main/java/org/trips/service_framework/notificationHandler/WhatsappNotificationHandler.java

```diff
@@ -0,0 +1,83 @@
+package org.trips.service_framework.notificationHandler;
+
+import lombok.NonNull;
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.stereotype.Service;
+import org.springframework.util.ObjectUtils;
+import org.trips.service_framework.clients.MercuryClient;
+import org.trips.service_framework.clients.request.NotificationRequest;
+import org.trips.service_framework.clients.response.NotificationResponse;
+import org.trips.service_framework.dtos.WhatsappNotificationRequest;
+
+import java.util.Objects;
+
+@Slf4j
+@Service
+@RequiredArgsConstructor
+public class WhatsappNotificationHandler implements NotificationHandler {
+    private final MercuryClient mercuryClient;
+
+    /**
+     * calls the mercury service along with the request data.
+     * returns error status if any exception.
+     *
+     * @param notificationRequest data required for the whatsapp notification
+     * @param <T>                 base class for the notification request
+     * @return notification response from mercury service
+     */
+    @Override
+    public <T extends NotificationRequest> NotificationResponse send(@NonNull T notificationRequest) {
+        WhatsappNotificationRequest request = (WhatsappNotificationRequest) notificationRequest;
+
+
+        try {
+            validateRequest(request);
+            log.info("Sending Email using clientCode {} with subject {} and Data {} by user {}", request.getClientCode(), request.getSubject(), request.getData(), request.getUser());
+
+            return mercuryClient.sendWhatsappMessage(request);
+        } catch (Exception e) {
+            return NotificationResponse.getErrorStatus(e.getMessage());
+        }
+    }
+
+    /**
+     * checks if the values required for the whatsapp notification is present.
+     *
+     * @param request data required for the whatsapp notification
+     */
+
+    private void validateRequest(WhatsappNotificationRequest request) {
+        String name = Strings.WHATSAPP_NOTIFICATION_REQUEST;
+        if (Objects.isNull(request.getData())) {
+            throw new IllegalArgumentException(String.format("%s: data is null", name));
+        }
+        if (ObjectUtils.isEmpty(request.getClientCode())) {
+            throw new IllegalArgumentException(String.format("%s: clientCode is empty", name));
+        }
+        if (Objects.isNull(request.getSubject())) {
+            throw new IllegalArgumentException(String.format("%s: notification subject is null", name));
+        }
+        if (Objects.isNull(request.getData())) {
```

Comment:
duplocate null check, already completed on line 52. Check for `request.getData().getData()` here.